corehq/messaging/tasks.py

```diff
@@ -52,7 +53,14 @@ def sync_case_chunk_for_messaging_rule(domain, case_id_chunk, rule_id):
         try:
             with CriticalSection([get_sync_key(case_id)], timeout=5 * 60):
                 _sync_case_for_messaging_rule(domain, case_id, rule_id)
-        except Exception:
+        except Exception as e:
+            if LOG_SMS_DUPLICATION.enabled(domain):
+                from corehq.apps.hqwebapp.utils import log_sms_duplicates_soft_assert
+                log_sms_duplicates_soft_assert(
+                    False,
+                    f'A duplicate SMS may or may not have been sent to '
+                    f'{case_id} due to exception {str(e)}'
```

Comment:
nit
```suggestion
                    f'{case_id} due to exception {e}'
```
Also, single-letter variables are not a best practice.

Reply:
I see `e` for an exception in python as pretty conventional (at least in sample code) if the block isn't too long.

Reply:
In my opinion that's not good sample code if it's doing that.

Single letter variables often make code harder to read. The only place where they are admissible according to our style conventions is in list comprehensions or generator expressions contained within a single line. It's not onerous to use a name like `error` or `exc` in an except block.

Reply:
`Single letter variables often make code harder to read. `

I agree with this in general except for in certain cases that are conventional, like using `i` in a for loop and `e` in a catch block. Both seem to transcend programming languages [Java](https://docs.oracle.com/javase/tutorial/essential/exceptions/catch.html, using `e` or `ex`) does this as well. I think for certain use cases like for and catch blocks, when you're reading the code you know you're in that block and what that var stands for. Agreed that if the block is too long, it can get confusing.

Reply:
I don't have a strong opinion about what's better, but I'll just note that this line is in pretty good company in this particular codebase (`-w` means match only along word boundaries):
```
(cchq) ➜  commcare-hq git:(master) ✗ ag -w 'as e' | wc
     482    2478   39377
(cchq) ➜  commcare-hq git:(master) ✗ ag -w 'as err' | wc
      85     442    7212
(cchq) ➜  commcare-hq git:(master) ✗ ag -w 'as error' | wc
      22     184    2203
(cchq) ➜  commcare-hq git:(master) ✗ ag -w 'as ex' | wc
      22     110    1761
(cchq) ➜  commcare-hq git:(master) ✗ ag -w 'as exc' | wc
       2      11     155
(cchq) ➜  commcare-hq git:(master) ✗ ag -w 'as exception' | wc
       1       5     101
```
"because we do it everywhere else" isn't always a good argument, but is important if we're going to go with the "it's a convention" argument.

Reply:
[Chapter 2 of Clean Code](https://duckduckgo.com/?q=clean+code+chapter+2) (dev book club read a few books ago) is be a more articulate source. My primary reasoning for preferring a longer name is searchability. Even a very marginally longer name like `ex` is better than `e` because e is such a common letter and can be hard to spot. I guess you could say `except ... as e` is a pet peeve of mine. However, I also use `i` occasionally in combination with `enumerate`, so I am certainly not perfect.

Fair point @dannyroberts, although we have a lot of code that was written before we had agreed on a consistent code style.

Reply:
wow, this really blew up. For the majority of our try/except blocks, I've always seen 
```
<error> as e: 
```
in the exception clause. Even in docs/wikis this pattern is referenced. And tbh, I would consider that NOT bad practice, as this is very clearly an error block and this is very much a common and familiar pattern! See https://wiki.python.org/moin/HandlingExceptions as one of many examples used in practice outside of our codebase.

I made sure to wrap `e` in `str()` as I've sometimes experienced errors being thrown in the past when I did not do this. Maybe that was fixed since python 3 with f strings. But as this code is temporary anyway...I just wanted to be REALLY careful to avoid causing additional errors in this part of the code.